nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Utilities/MSBuildHelper.cs

```diff
@@ -341,10 +343,34 @@ private static ProjectRootElement CreateProjectRootElement(ProjectBuildFile buil
         var projectDirectory = Path.GetDirectoryName(projectPath);
         projectDirectory ??= repoRoot;
         var topLevelFiles = Directory.GetFiles(repoRoot);
-        var nugetConfigPath = PathHelper.GetFileInDirectoryOrParent(projectDirectory, repoRoot, "NuGet.Config", caseSensitive: false);
+        var nugetConfigPath = PathHelper.GetFileInDirectoryOrParent(projectPath, repoRoot, "NuGet.Config", caseSensitive: false);
         if (nugetConfigPath is not null)
         {
+            var nugetConfigDir = Path.GetDirectoryName(nugetConfigPath);
```

Comment:
This seems very NuGet-y. Should we extract this if body to the NuGetHelpers class?

Reply:
We spoke offline and agree that if we split this up it would be within MSBuildHelper, and only if another iteration is needed anyway.